vtex/utils/transform.ts

```diff
@@ -486,14 +486,29 @@ const toBreadcrumbList = (
   };
 };
 
-const legacyToProductGroupAdditionalProperties = (product: LegacyProductVTEX) =>
-  product.allSpecifications?.flatMap((name) => {
-    const values = (product as unknown as Record<string, string[]>)[name];
+const legacyToProductGroupAdditionalProperties = (
+  product: LegacyProductVTEX,
+) => {
+  const groups = product.allSpecificationsGroups ?? [];
+  const specifications = product.allSpecifications ?? [];
+
+  const groupMappings = groups.flatMap((group) => {
+    const groupValues = (product as unknown as Record<string, string[]>)[group];
+    return groupValues.map((value) => ({ name: value, isFrom: group }));
+  });
 
+  return specifications.flatMap((spec) => {
+    const values = (product as unknown as Record<string, string[]>)[spec];
     return values.map((value) =>
-      toAdditionalPropertySpecification({ name, value })
+      toAdditionalPropertySpecification({
+        name: spec,
+        value,
+        additionalType: groupMappings.find((mapping) => mapping.name === spec)
```

Comment:
https://github.com/deco-cx/apps/pull/645

Can you check this other PR? We add this info in other prop called propertyID. It would be nice keep the same on both functions.